mypy/semanal.py

```diff
@@ -740,8 +744,10 @@ def leave_class(self) -> None:
         self.locals.pop()
         self.type = self.type_stack.pop()
 
-    def analyze_class_decorator(self, defn: ClassDef, decorator: Expression) -> None:
+    def analyze_class_decorator(self, defn: ClassDef, decorator: Expression) -> bool:
         decorator.accept(self)
+        return (isinstance(decorator, RefExpr) and
+                decorator.fullname in ('typing.runtime', 'typing_extensions.runtime'))
```

Comment:
Move this to a separate helper method instead of hijacking ``analyze_class_decorator``.